dvc/stage/loader.py

```diff
@@ -42,7 +42,9 @@ def fill_from_lock(stage, lock_data=None):
             if isinstance(item, dependency.ParamsDependency):
                 item.fill_values(get_in(lock_data, [stage.PARAM_PARAMS, path]))
                 continue
-            item.checksum = get_in(checksums, [key, path, item.checksum_type])
+            item.info = get_in(checksums, [key, path], {})
+            item.info = item.info.copy()
+            item.info.pop("path", None)
```

Comment:
For the record: Discussed with @skshetry that we really need to move serialization/loading to specific deps/outputs (e.g. in a form of custom loaders/dumpers). So this is a temporary workaround.